cumulusci/cli/cci.py

```diff
@@ -216,25 +167,68 @@ def new_func(*args, **kw):
         return decorate(func)
 
 
-# Root command
+SUGGEST_GIT_GIST_COMMAND = """\nIt looks like you may have run into error. Did you know cci has a command for sending this error to a GitHub gist?
+Just run `$ cci gist` and make sure that your GitHub access token has the 'create gist' scope.\n"""
 
 
-@click.group("main", help="")
-def main():
+#
+# Root command
+#
+def main(args=None):
     """Main CumulusCI CLI entry point.
 
     This runs as the first step in processing any CLI command.
+
+    This wraps the `click` library in order to do some initialization and centralized error handling.
     """
-    # Avoid checking for updates if we've been asked to output JSON,
-    # or if we're going to check anyway as part of the `version` command.
-    is_version_command = len(sys.argv) > 1 and sys.argv[1] == "version"
-    if "--json" not in sys.argv and not is_version_command:
-        check_latest_version()
-    log_requests = "--debug" in sys.argv
-    init_logger(log_requests=log_requests)
+    with contextlib.ExitStack() as stack:
+        args = args or sys.argv
+        # Check for updates _unless_ we've been asked to output JSON,
+        # or if we're going to check anyway as part of the `version` command.
+        is_version_command = len(args) > 1 and args[1] == "version"
+        if "--json" not in args and not is_version_command:
+            check_latest_version()
+
+        # Load CCI config
+        global RUNTIME
+        RUNTIME = CliRuntime()
+        RUNTIME.check_cumulusci_version()
+
+        # Configure logging
+        debug = "--debug" in args
+        if debug:
+            args.remove("--debug")
+
+        # Only create logfiles for commands
+        # that are not `cci gist`
+        is_gist_command = len(args) > 1 and args[1] == "gist"
+        if not is_gist_command:
+            logger = get_gist_logger()
+            stack.enter_context(tee_stdout_stderr(args, logger))
+
+        init_logger(log_requests=debug)
+        # Hand CLI processing over to click, but handle exceptions
+        try:
+            cli(standalone_mode=False)
+        except Exception as e:
+            # Display the error
+            if debug:
+                traceback.print_exc()
+                pdb.post_mortem()
+            else:
+                click.echo(click.style(f"Error: {e}", fg="red"))
+                click.echo(click.style(SUGGEST_GIT_GIST_COMMAND, fg="yellow"))
```

Comment:
@Br4nd0R we should make sure the traceback gets logged to the file even when it's not printed to stdout/stderr (i.e. even when the `--debug` flag is not set)